### PR TITLE
move frontend to separate deployment

### PR DESCRIPTION
**Pull request by quinn-freshly**

### CODE REVIEW

File: .tuber/virtualservice.yaml [hunk 1]
```diff
@@ -13,6 +13,14 @@ spec:
             host: tuber
             port:
               number: 3000
+      match:
+        - uri:
+            prefix: /tuber/graphql
+    - route:
+        - destination:
+            host: tuber-frontend.tuber-frontend.svc.cluster.local
```
Comment (jkminneti) [inline .tuber/virtualservice.yaml:21]: new virtualservice inside of tuber-frontend should handle this

with `{{.tuberAppName}}`'s all the way down

question then becomes how to istio it. can we exact match on `/tuber/graphql` in tuber-frontend, and prefix match on `/tuber`, and have it just work?

or will we have to exact match on `/tuber`, and prefix match on `/tuber/apps` etc

either way should be doable

why? FRONTEND REVIEW APPS 🎉

File: cloudbuild.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+steps:
```
Comment (jkminneti) [inline cloudbuild.yaml:1]: should split into two cloudbuild yamls, one for frontend one for backend, _both using kaniko..._, and with file filters to match (assuming, hopefully safely, we can segment frontend by a toplevel dir)

File: pkg/adminserver/web/.tuber/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: tuber-frontend
```
Comment (jkminneti) [inline pkg/adminserver/web/.tuber/deployment.yaml:4]: if we do go the frontend review apps route, this should all be `{{.tuberAppName}}`'d (same for the service)

File: pkg/iap/auth.go [hunk 4]
```diff
@@ -32,7 +31,6 @@ func init() {
 }
 
 func RefreshTokenExists() bool {
-	spew.Dump(refreshTokenFile)
```
Comment (jkminneti) [inline pkg/iap/auth.go:35]: 🙏 🙏 🙏 🔪

File: .tuber/persistentvolumeclaim.yaml [hunk 5]
```diff
@@ -2,7 +2,6 @@ apiVersion: v1
 kind: PersistentVolumeClaim
 metadata:
   name: tuber
-  namespace: tuber
```
Comment (jkminneti) [inline .tuber/persistentvolumeclaim.yaml:5]: these were there to _explicitly disallow review apps of tuber_

tuber now has a `TuberApp.ReviewAppsConfig.Enabled = false` but stilllllll idk up to you.

**Review by jkminneti — CHANGES_REQUESTED** [review]

File: .tuber/virtualservice.yaml (hunk 1 above)
Comment (jkminneti) [reply .tuber/virtualservice.yaml:21]: actually you probably use relative routing to get to `/tuber/graphql`

so if a review app is running on `/tuber-pops-123`, we'll need another route to define `/tuber-pops-123/graphql`

which means `tuber` doesn't even need a virtualservice!

File: cloudbuild.yaml (hunk 2 above)
Comment (quinn-freshly) [reply cloudbuild.yaml:1]: idk how 2 kaniko

File: .tuber/persistentvolumeclaim.yaml (hunk 5 above)
Comment (quinn-freshly) [reply .tuber/persistentvolumeclaim.yaml:5]: ehh, lets just use the setting now that we have it.

**Review by jkminneti — APPROVED** [review]
fixes the route

**Pull request merged**